cura/Machines/Models/BaseMaterialsModel.py

```diff
@@ -95,40 +96,59 @@ def setEnabled(self, enabled):
                 self._update()
             self.enabledChanged.emit()
 
-    @pyqtProperty(bool, fset=setEnabled, notify=enabledChanged)
+    @pyqtProperty(bool, fset = setEnabled, notify = enabledChanged)
     def enabled(self):
         return self._enabled
 
+    ##  Triggered when a list of materials changed somewhere in the container
+    #   tree. This change may trigger an _update() call when the materials
+    #   changed for the configuration that this model is looking for.
+    def _materialsListChanged(self, material: MaterialNode) -> None:
+        if material.variant.container_id != self._extruder_stack.variant.getId():
+            return
+        global_stack = cura.CuraApplication.CuraApplication.getInstance().getGlobalContainerStack()
+        if not global_stack:
+            return
+        if material.variant.machine.container_id != global_stack.definition.getId():
+            return
+        self._update()
+
     ## This is an abstract method that needs to be implemented by the specific
     #  models themselves.
     def _update(self):
-        pass
+        self._favorite_ids = set(cura.CuraApplication.CuraApplication.getInstance().getPreferences().getValue("cura/favorite_materials").split(";"))
+
+        # Update the available materials (ContainerNode) for the current active machine and extruder setup.
+        global_stack = cura.CuraApplication.CuraApplication.getInstance().getGlobalContainerStack()
+        extruder_stack = global_stack.extruders.get(str(self._extruder_position))
+        if not extruder_stack:
+            return
+        nozzle_name = extruder_stack.variant.getName()
+        materials = ContainerTree.getInstance().machines[global_stack.definition.getId()].variants[nozzle_name].materials
+        compatible_material_diameter = str(round(extruder_stack.getCompatibleMaterialDiameter()))
```

Comment:
Same as above, maybe use `getApproximateMaterialDiameter()` and rename `compatible_material_diameter` to `approximate_material_diameter`?

Reply:
Agreed. Done: 5f4051ab5c62d19313e29c8207a9ebaf6e8b2192